icu4c/source/common/locdispnames.cpp

```diff
@@ -384,6 +384,13 @@ _getDisplayNameForComponent(const char *locale,
 
     root = tag == _kCountries ? U_ICUDATA_REGION : U_ICUDATA_LANG;
 
+    if (displayLocale != NULL && (*displayLocale == '\0' || *displayLocale == '_')) {
+        // Magic string that will have the result that the resource loading code
```

Comment:
Normally, the displayLocale is passed into ures_open() to load a bundle, and then we look for the right data in that bundle (and its parent chain). The ures_open() implementation should do the right thing, and I remember that one of your recent changes had a fix that I though should handle _IN -> default. We should not need any special handling in this higher-level code.